### PR TITLE
feat: Convert logger module to TS

### PR DESCRIPTION
**Pull request by yavorona**
## Summary
- Convert `index.js` and `index.react_native.js`to TS
- Remove `enums.js` since it is not consumed by any module

## Test plan
Existing unit tests

### CODE REVIEW

File: packages/optimizely-sdk/lib/core/bucketer/index.tests.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ import {
   LOG_MESSAGES,
   LOG_LEVEL,
 } from '../../utils/enums';
-import logger from '../../plugins/logger';
+import * as logger from '../../plugins/logger';
```
Comment (zashraf1985) [inline packages/optimizely-sdk/lib/core/bucketer/index.tests.js:27]: Does importing by name work?

File: packages/optimizely-sdk/lib/plugins/logger/index.ts [hunk 2]
```diff
@@ -13,21 +13,24 @@
...
+type ConsoleLogHandlerConfig = {
+  logLevel?: LogLevel | string;
+  logToConsole?: boolean;
+  prefix?: string;
+}
 
-NoOpLogger.prototype.log = function() {};
+export class NoOpLogger {
+  log(): void {
+    () => {};
```
Comment (zashraf1985) [inline packages/optimizely-sdk/lib/plugins/logger/index.ts:26]: can this be removed?

**Review by zashraf1985 — CHANGES_REQUESTED** [review]

File: packages/optimizely-sdk/lib/core/bucketer/index.tests.js (hunk 1 above)
Comment (yavorona) [reply packages/optimizely-sdk/lib/core/bucketer/index.tests.js:27]: good point, imported as individual export

File: packages/optimizely-sdk/lib/plugins/logger/index.ts (hunk 2 above)
Comment (yavorona) [reply packages/optimizely-sdk/lib/plugins/logger/index.ts:26]: `NoOpLogger`is consumed by react-native logger, not sure if it is a good idea to remove it

Comment (zashraf1985) [reply packages/optimizely-sdk/lib/plugins/logger/index.ts:26]: Oh no. I m just talking about the empty arrow function wrapped inside the log function. Think that extra empty arrow function might be redundant

Comment (yavorona) [reply packages/optimizely-sdk/lib/plugins/logger/index.ts]: Oh that is true, it is not needed here and is now removed

File: packages/optimizely-sdk/lib/plugins/logger/index.react_native.ts [hunk 3]
```diff
@@ -31,38 +32,30 @@ function getLogLevelName(level) {
...
-export var createNoOpLogger = function() {
+export function createNoOpLogger(): NoOpLogger {
   return new NoOpLogger();
 }
-
-export default {
-  createLogger,
-  createNoOpLogger,
-};
+
```
Comment (zashraf1985) [inline packages/optimizely-sdk/lib/plugins/logger/index.react_native.ts:61]: I dont know why github is showing `no newline at end of file` error even though it appears that there is a new line there. Can you check.

File: packages/optimizely-sdk/lib/plugins/logger/index.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright 2016-2017, 2020, Optimizely
+ * Copyright 2021, Optimizely
```
Comment (zashraf1985) [inline packages/optimizely-sdk/lib/plugins/logger/index.ts:2]: Shouldn't this include previous years too. Technically, this is a new file but logically this is a rename. Just a suggestion, feel free to do whatever makes sense to you here

**Review by zashraf1985 — APPROVED** [review]
LGTM! Approved with a few optional suggestions

**Pull request merged**